### PR TITLE
server: Add PlayFull method to override options

### PR DESCRIPTION
**Pull request by cfoch**
https://phabricator.endlessm.com/T24426

### CODE REVIEW

File: src/server.py [hunk 1]
```diff
@@ -118,26 +123,40 @@ def _ensure_release_countdown(self):
...
+            options = {}
+
         self._cancel_countdown()
         self.hold()
 
         uuid_ = str(uuid.uuid4())
-        self.players[uuid_] = HackSoundPlayer(self.metadata[sound_event_id],
-                                              sender)
+        metadata = self.metadata[sound_event_id].copy()
+        metadata.update(options)
```
Comment (ptomato) [inline src/server.py]: We can't do it this way; if, for example, a pitch of 1.5 is specified in the metadata file, and we specify pitch 1.06 in the options, then we want the pitch to become higher, not lower.

Volumes and pitches should be multiplied, delays should be added, and fade-in/fade-out times should probably be overridden.

Comment (cfoch) [reply src/server.py]: Should that multiplications be accumulated? If you call 3 times PlayFull with a pitch 1.06, should the pitch be multiplied by 1.06 * 1.06 * 1.06 in the third call?

Comment (ptomato) [reply src/server.py]: No, the parameters in PlayFull should only take effect for that one instance of playing the sound. So if you call it three times with pitches of 1.5, 1.4, 1.3, then the pitches should get subsequently lower, not higher.

**Comment by cfoch** [comment]
Depends on #24

File: src/server.py [hunk 2]
```diff
@@ -257,26 +266,35 @@ def _ensure_release_countdown(self):
...
-                                              sender)
+        metadata = self.metadata[sound_event_id]
+        self.players[uuid_] = HackSoundPlayer(metadata, sender, options)
         self.players[uuid_].connect("eos", self.__player_eos_cb, uuid_)
         self.players[uuid_].connect("error", self.__player_error_cb, uuid_,
                                     connection, path, iface)
         self.players[uuid_].play()
         invocation.return_value(GLib.Variant('(s)', (uuid_, )))
 
+    def play_sound(self, sound_event_id, connection, sender, path, iface,
```
Comment (ptomato) [inline src/server.py]: I would remove these and just call `self.play_sound()` directly in both the `PlaySound` and `PlayFull` cases below.

**Comment by cfoch** [comment]
I am removing the speed support in the PlayFull method to remove dependency on #24

**Pull request merged**